### PR TITLE
feat(core): kas-grants list

### LINKED ISSUE #253: KAS grants does not support list

### LINKED ISSUE #332: Test e2e: KAS Grants LIST
### Background

We recently added support for BATS e2e tests, with a few test runs currently in the pipeline. We need to add e2e tests for the critical paths within policy:
`kas-grants list`

### Acceptance Criteria
- [ ] list is e2e tested in current `kas-grants.bats` file

Blocked by #253

### PR DESCRIPTION
**Pull request by jakedoublev**
Resolves #332 
Resolves #253

### CODE REVIEW

**Comment by jakedoublev** [comment]
Screenshot 2024-08-29 at 1 19 14 PM

**Comment by jakedoublev** [comment]
Blocked by https://github.com/opentdf/platform/pull/1478

File: cmd/kas-grants.go [hunk 1]
```diff
@@ -153,6 +155,80 @@ func policy_unassignKasGrant(cmd *cobra.Command, args []string) {
...
+	if kasF != "" {
+		_, err := uuid.Parse(kasF)
+		if err != nil {
+			kasURI = kasF
+		} else {
+			kasID = kasF
+		}
+	}
+
+	grants, err := h.ListKasGrants(cmd.Context(), kasID, kasURI)
```
Comment (suchak1) [inline cmd/kas-grants.go:178]: Does this return all kas grants in the db if `kasF` is an empty string? Just verifying that sure the `--kas` flag is optional.

Comment (jakedoublev) [reply cmd/kas-grants.go:178]: Yes, precisely. You can see that demonstrated here in the e2e tests added for this functionality: https://github.com/opentdf/otdfctl/pull/346/files#diff-53724192e0c3172f28c28158e89035dd43c5358ca19a4d322a3072050af48c43R49-R66

**Review by suchak1 — APPROVED** [review]
lgtm

**Pull request merged**